src/main/kotlin/io/sip3/salto/ce/rtpr/RtprHandler.kt

```diff
@@ -0,0 +1,175 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.salto.ce.rtpr
+
+import io.micrometer.core.instrument.DistributionSummary
+import io.micrometer.core.instrument.Metrics
+import io.micrometer.core.instrument.Timer
+import io.netty.buffer.Unpooled
+import io.sip3.commons.domain.payload.RtpReportPayload
+import io.sip3.commons.util.format
+import io.sip3.salto.ce.RoutesCE
+import io.sip3.salto.ce.USE_LOCAL_CODEC
+import io.sip3.salto.ce.domain.Packet
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.json.JsonObject
+import mu.KotlinLogging
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.TimeUnit
+
+open class RtprHandler : AbstractVerticle() {
+
+    private val logger = KotlinLogging.logger {}
+
+    companion object {
+
+        const val PREFIX_RTP = "rtpr_rtp"
+        const val PREFIX_RTCP = "rtpr_rtcp"
+
+        const val JITTER = "_jitter"
+        const val R_FACTOR = "_r-factor"
+        const val MOS = "_mos"
+        const val EXPECTED_PACKETS = "_expected-packets"
+        const val LOST_PACKETS = "_lost-packets"
+        const val REJECTED_PACKETS = "_rejected-packets"
+
+        const val DURATION = "_duration"
+    }
+
+    private var timeSuffix: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMdd")
+
+    override fun start() {
+        config().getString("time-suffix")?.let {
+            timeSuffix = DateTimeFormatter.ofPattern(it)
+        }
+
+        vertx.eventBus().localConsumer<Packet>(RoutesCE.rtpr) { event ->
+            try {
+                val packet = event.body()
+                handle(packet)
+            } catch (e: Exception) {
+                logger.error("RtprHandler 'handle()' failed.", e)
+            }
+        }
+    }
+
+    open fun handle(packet: Packet) {
+        val payload = Unpooled.wrappedBuffer(packet.payload)
+        val report = RtpReportPayload().apply { decode(payload) }
+        val prefix = prefix(report.source)
+
+        writeToDatabase(prefix, packet, report)
+        calculateMetrics(prefix, packet, report)
+    }
+
+    open fun prefix(source: Byte): String {
+        return when (source) {
+            RtpReportPayload.SOURCE_RTP -> PREFIX_RTP
+            RtpReportPayload.SOURCE_RTCP -> PREFIX_RTCP
+            else -> throw IllegalArgumentException("Unsupported RTP report source: '${source}'")
+        }
+    }
+
+    open fun calculateMetrics(prefix: String, packet: Packet, report: RtpReportPayload) {
+        // Skip cumulative reports
+        if (report.cumulative) {
```

Comment:
For some reason I think that it makes sense to write cumulative reports only. In this case we will have a better picture comparing CAPs from SIP signaling and QoS data coming from RTP.

Reply:
Agree, let's have it as an option to choose